lib/build-components/build_components/templates/python/macros.j2

```diff
@@ -0,0 +1,180 @@
+{# [Builder image]
+    Download and install poetry under the root user.
+#}
+{% macro install_poetry(version) %}
+# Install poetry
+# Note: We prefer to install python packages as --user so that the artifacts are easier
+#       to copy out of the builder image into the final image. poetry already installs
+#       itself under the user directory, and we eventually copy the poetry tool and the
+#       installed python packages under the image USER in the resulting image.
+
+# Ensure we have curl so we can install poetry
+RUN apt-get update \
+ && apt-get -y --no-install-recommends install \
+    curl
+
+# Configure pip, mainly to ensure that packages are installed in the user directory.
+COPY pip.conf /etc/pip.conf
+
+# Install poetry under the root user's home directory.
+# On some images "sh" is aliased to "dash" which does not support "set -o pipefail".
+# We use the "exec" form of RUN to delegate this command to bash instead.
+# This is all because we have a pipe in this command and we wish to fail the build
+# if the any command in the pipeline fails.
+RUN ["/bin/bash", "-c", "set -o pipefail && pip install --upgrade pip && curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | POETRY_VERSION={{version}} python && chmod a+x /root/.poetry/bin/poetry"]
+COPY poetry.toml /root/.config/pypoetry/config.toml
+ENV PATH=/root/.poetry/bin:$PATH
+{% endmacro %}
+
+
+{# [Builder image]
+    Install or copy the component.
+#}
+{% macro build_component(component, build_info) %}
+FROM {{build_info.builder_image}} AS builder-{{component}}
+
+# Install packages required to build native library components
+RUN apt-get update \
+ && apt-get -y --no-install-recommends install \
+    build-essential
```

Comment:
why split across two lines?

Reply:
meh, nearly every other `apt-get ... install` we use puts all the packages on their own lines. Just looks a little weird when there's just one package.